imports/ui/components/study_plan/ModuleSearchBar.jsx

```diff
@@ -0,0 +1,210 @@
+  //   this.setState({
+  //     userInput: event.target.value
+  //   });
+  // }
+
+  updateModuleInput(event, {newValue}) {
+    console.log("updateModuleInput event: " + event +
+                " newValue: " + newValue);
+    this.setState({
+      userInput: newValue
+    });
+  }
+
+  render() {
+    const value = this.state.userInput;
+    const suggestions = this.state.suggestions;
+    const inputProps = {
+      placeholder: 'Module code',
+      value,
+      onChange: this.updateModuleInput.bind(this)
+    };
+
+    return (
+      <Autosuggest
+        id={this.props.id}
+        inputProps={inputProps}
+        suggestions={suggestions}
+        onSuggestionsFetchRequested={this.onSuggestionsFetchRequested.bind(this)}
+        onSuggestionsClearRequested={this.onSuggestionsClearRequested.bind(this)}
+        getSuggestionValue={getSuggestionValue}
+        renderSuggestion={renderSuggestion}
+        renderSuggestionsContainer={renderSuggestionsContainer}
+        focusFirstSuggestion={true}
+        focusInputOnSuggestionClick={false}
+      />
+    );
+  }
+}
+
+Module.propTypes = {
+  // ID that identifies this component uniquely.
+  // This id also uniquely defines the Autosuggest box - necessary if there are
+  // many Autosuggest boxes on the screen.
+  id: React.PropTypes.string
+}
+
+//======================================================
+// Helper functions for configuring the Autosuggest React component
+//======================================================
+
+const languages = [
+  {
+    code: 'adrian',
+    year: 1972
+  },
+  {
+    code: 'benedict',
+    year: 2012
+  },
+  {
+    code: 'charles',
+    year: 1972
+  },
+  {
+    code: 'charlie',
+    year: 2012
+  },
+  {
+    code: 'elmer',
+    year: 1972
+  },
+  {
+    code: 'emmanuel',
+    year: 2012
+  },
+  {
+    code: 'elaine',
+    year: 1972
+  },
+  {
+    code: 'honey',
+    year: 2012
+  },
+  {
+    code: 'homer',
+    year: 1972
+  },
+  {
+    code: 'heath',
+    year: 2012
+  },
+  {
+    code: 'hebrew',
+    year: 1972
+  },
+  {
+    code: 'helberg',
+    year: 2012
+  },
+  {
+    code: 'heinz',
+    year: 1972
+  },
+  {
+    code: 'hubert',
+    year: 2012
+  }
+]
+
+/**
+ * Defines how suggestions are displayed as a list to the user.
+ *
+ * @param  {[string]} value    Value input by the user in the autocomplete
+ *                             input box
+ * @return {[array]}    List of suggested modules
+ */
+const getSuggestions = (userInput) => {
+  console.log("getSuggestions: " + userInput);
+  let inputValue = userInput.value;
+  inputValue = inputValue.trim().toLowerCase(); // Perform string normalization
+  const inputLength = inputValue.length;
+
+  if (inputLength === 0) {
+    // No suggestions given to user if they don't type in anything
+    return [];
+  } else {
+    // Return elements in the list with beginning letters that match inputValue
+    return languages.filter((lang) => {
```

Comment:
Noted! Yes indeed :) I intend to do an async call here, and when changes are made to the state of the component, it'll automatically be re-rendered!